src/pip/_internal/req/req_install.py

```diff
@@ -394,14 +394,33 @@ def move_to_correct_build_directory(self):
         This is only called to "fix" the build directory after generating
         metadata.
         """
+        assert self.req is None
+        assert self.metadata is not None
+
+        # Construct a Requirement object from the generated metadata
+        if isinstance(parse_version(self.metadata["Version"]), Version):
+            op = "=="
+        else:
+            op = "==="
+
+        self.req = Requirement(
```

Comment:
nit: this method now does a little more than its name `move_to_correct_build_directory` ^^
Maybe `normalize_req_and_build_directory` ? (I'm not convinced :stuck_out_tongue: )

Reply:
Yea, is it okay if I change that in a follow up PR? I have one ready to go -- just waiting on a "this looks OK" for this PR to merge and file that.

Reply:
(also, thanks for taking a look! ^>^)